turn/src/allocation/allocation_manager.rs

```diff
@@ -41,6 +41,33 @@ impl Manager {
         Ok(())
     }
 
+    // Returns the information about the all [`Allocation`]s associated with
+    // the specified [`FiveTuple`]s.
+    pub async fn get_allocations_info(
+        &self,
+        five_tuples: Option<Vec<FiveTuple>>,
+    ) -> HashMap<FiveTuple, AllocationInfo> {
+        let mut infos = HashMap::new();
+
+        let guarded = self.allocations.lock().await;
+
+        guarded.iter().for_each(|(five_tuple, alloc)| {
+            if five_tuples.is_none() || five_tuples.as_ref().unwrap().contains(five_tuple) {
+                #[cfg(not(feature = "metrics"))]
+                let relayed_bytes: Option<usize> = None;
+                #[cfg(feature = "metrics")]
```

Comment:
Why not just feature gate this whole method? There's no value in `AllocationInfo` if not using metrics, no?

Reply:
No, it will return `AllocationInfo` with all the other information and `relayed_bytes: None` which indicates that metrics have not been gathered.